ospfd/ospf_sr.c

```diff
@@ -636,7 +618,25 @@ static int ospf_zebra_send_mpls_labels(int cmd, struct sr_nhlfe nhlfe)
 			   inet_ntoa(nhlfe.prefv4.prefix),
 			   nhlfe.prefv4.prefixlen, nhlfe.ifindex);
 
-	return zclient_send_message(zclient);
+	zl.type = ZEBRA_LSP_OSPF_SR;
+	zl.local_label = nhlfe.label_in;
+
+	SET_FLAG(zl.message, ZAPI_LABELS_FTN);
+	zl.route.prefix.family = nhlfe.prefv4.family;
+	zl.route.prefix.prefixlen = nhlfe.prefv4.prefixlen;
+	zl.route.prefix.u.prefix4 = nhlfe.prefv4.prefix;
+	zl.route.type = ZEBRA_ROUTE_OSPF;
+	zl.route.instance = 0;
+
+	zl.nexthop_num = 1;
+	znh = &zl.nexthops[0];
+	znh->type = NEXTHOP_TYPE_IPV4_IFINDEX;
+	znh->family = AF_INET;
+	znh->address.ipv4 = nhlfe.nexthop;
+	znh->ifindex = nhlfe.ifindex;
+	znh->label = nhlfe.label_out;
+
+	return zebra_send_mpls_labels(zclient, cmd, &zl);
 }
 
 /* Request zebra to install/remove FEC in FIB */
```

Comment:
Why do you have not updated `ospf_zebra_send_mpls_ftn()` function with the new ZAPI message ? In fact, with the new ZAPI message, both `ospf_zebra_send_mpls_labels()` & `ospf_zebra_send_mpls_ftn()` functions could be merge into a single call to ZEBRA. Please, update accordingly.

Reply:
Indeed. The new API should also fix ECMP in ospfd when SR is used. I'll fix this now.